app/eventyay/presale/templates/pretixpresale/fragment_modals.html

```diff
@@ -14,4 +14,374 @@ <h3></h3>
+    display: flex;
+    align-items: center;
+    justify-content: center;
+}
+
+.checkout-login-popup {
+    background: white;
+    border-radius: 8px;
+    max-width: 420px;
+    width: 90%;
+    max-height: 90vh;
+    overflow-y: auto;
+    box-shadow: 0 4px 20px rgba(0, 0, 0, 0.3);
+}
+
+.checkout-login-header {
+    display: flex;
+    justify-content: space-between;
+    align-items: center;
+    padding: 20px;
+    border-bottom: 1px solid #eee;
+}
+
+.checkout-login-header h3 {
+    margin: 0;
+    font-size: 1.4em;
+}
+
+.checkout-login-close {
+    background: none;
+    border: none;
+    font-size: 28px;
+    cursor: pointer;
+    color: #666;
+    padding: 0;
+    line-height: 1;
+}
+
+.checkout-login-close:hover {
+    color: #333;
+}
+
+.checkout-login-body {
+    padding: 20px;
+}
+
+.checkout-login-message {
+    text-align: center;
+    color: #666;
+    margin-bottom: 20px;
+}
+
+.checkout-login-divider {
+    text-align: center;
+    margin: 20px 0;
+    position: relative;
+}
+
+.checkout-login-divider::before {
+    content: '';
+    position: absolute;
+    left: 0;
+    top: 50%;
+    width: 100%;
+    height: 1px;
+    background: #ddd;
+}
+
+.checkout-login-divider span {
+    background: white;
+    padding: 0 15px;
+    position: relative;
+    color: #999;
+}
+
+.checkout-login-sso .btn {
+    margin-bottom: 10px;
+}
+
+.checkout-login-footer {
+    text-align: center;
+    margin-top: 20px;
+    padding-top: 15px;
+    border-top: 1px solid #eee;
+    color: #666;
+}
+
+.checkout-login-footer a {
+    color: #337ab7;
+}
+</style>
+
+<script type="module">
+const modal = document.getElementById('checkout-login-modal');
+if (modal) {
+    const loginBtn = document.getElementById('checkout-login-btn');
+    const closeBtn = modal.querySelector('.checkout-login-close');
+    const loginForm = document.getElementById('checkout-login-form');
+    const errorDiv = document.getElementById('checkout-login-error');
+    const nextUrlInput = document.getElementById('checkout-next-url');
+    let checkoutUrl = '';
+
+    // Constants for SSO popup
+    const POPUP_WIDTH = 600;
+    const POPUP_HEIGHT = 700;
+    const POPUP_CHECK_INTERVAL = 500;
+    let activeCheckLogin = null;
+
+    // URL parameter name for redirect after login/register
+    const REDIRECT_PARAM = 'next';
+
+    // Open modal
+    if (loginBtn) {
+        loginBtn.addEventListener('click', (e) => {
+            e.preventDefault();
+            checkoutUrl = loginBtn.dataset.checkoutUrl || '';
+            if (nextUrlInput) nextUrlInput.value = checkoutUrl;
+
+            // Update SSO links with next URL
+            const ssoLinks = modal.querySelectorAll('.checkout-sso-btn');
+            ssoLinks.forEach((link) => {
+                const baseUrl = link.href.split('?')[0];
+                const backend = link.dataset.provider;
+                link.href = `${baseUrl}?backend=${backend}&${REDIRECT_PARAM}=${encodeURIComponent(checkoutUrl)}`;
```

Comment:
It is cleaner if you build the whole query string with `URLSearchParams`.
By the way, you should check if `checkoutUrl` is non-empty before adding "next" param.